core/commands/storage.go

```diff
@@ -304,20 +312,31 @@ var storageUploadRequestChallengeCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
-		cid, err := cid2.Parse(chunkHash)
+		cid, err := cidlib.Parse(chunkHash)
 		if err != nil {
 			return err
 		}
-		sch, err := NewStorageChallenge(req.Context, n, api, cid)
-		if err != nil {
-			return err
+
+		var mutex = sync.Mutex{}
+		mutex.Lock()
+		defer mutex.Unlock()
+
+		var sch *StorageChallenge
+		// in multi-process taling to mulltiple hosts, different cids can only generate one storage challenge
+		// if existing challenge with the same cid, means this challenge has been generated and sent
+		if sch == nil || sch.ID != cid.String() {
```

Comment:
Why are you comparing the challenge ID with a chunk ID?? They are two different kinds of IDs